hrms/hr/doctype/leave_application/leave_application.js

```diff
@@ -105,6 +105,9 @@ frappe.ui.form.on("Leave Application", {
 		}
 
 		frm.trigger("set_employee");
+		if (frm.doc.employee) {
```

Comment:
```suggestion
		if (frm.doc.docstatus === 0) {
```
make_dashboard already checks for employee value so removing the condition here. Also, the dashboard should only be shown on draft transactions. Doesn't make sense to fetch the current status for old transactions/already submitted ones.